### PR TITLE
[NG] MOVE SofaComponent* to packages

### PR DESCRIPTION
**Pull request by guparan**
Please gently welcome this first NG pull-request :blush: 

The base idea was to remove the SofaComponent* packages from modules and to put them in their own place instead.
Doing this, I realized that SofaComponentBase contains actual components (it's not only a basic package as it should be).
Thus, it permitted me to introduce the first NG module Sofa.Component.Utils and fill it with SofaComponentBase intruders (MakeAliasComponent, MakeDataAliasComponent, MessageHandlerComponent, InfoComponent).

NG architecture is based on the discussions in #543.
Transition headers are also provided (in deprecated_layout folder) to ensure old includes.

I am also working with @damienmarchal on a way to cleanly automatize this kind of NG refactoring with python: see https://github.com/guparan/sofa2ng

Feedback is more than welcome!

______________________________________________________

This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [x] does not generate new unit test failures.
- [x] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by guparan** [comment]
[ci-build][with-scene-tests]

File: modules/Sofa/Component/Utils/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+project(Sofa.Component.Utils VERSION 1.0)
+
+find_package(SofaFramework REQUIRED)
+
+set(HEADER_FILES config/${PROJECT_NAME}.h)
+set(SOURCE_FILES config/${PROJECT_NAME}.cpp)
+
+list(APPEND HEADER_FILES
+    deprecated_layout/SofaComponentBase/MakeAliasComponent.h
```
Comment (damienmarchal) [inline modules/Sofa/Component/Utils/CMakeLists.txt:11]: Please validate deprecated_layout principle

File: modules/Sofa/Component/Utils/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    deprecated_layout/SofaComponentBase/MakeDataAliasComponent.h
+    deprecated_layout/SofaComponentBase/MessageHandlerComponent.h
+    deprecated_layout/SofaComponentBase/InfoComponent.h
+    )
+
+list(APPEND HEADER_FILES
+    src/sofa/component/utils/MakeAliasComponent.h
```
Comment (damienmarchal) [inline modules/Sofa/Component/Utils/CMakeLists.txt:18]: Please validate src/namespace/component principle

File: applications/sofa/gui/qt/CMakeLists.txt [hunk 3]
```diff
@@ -1,6 +1,8 @@
 cmake_minimum_required(VERSION 3.1)
 project(SofaGuiQt)
 
+find_package(Sofa.Component.Utils)
```
Comment (damienmarchal) [inline applications/sofa/gui/qt/CMakeLists.txt:4]: Please give feedback on "." naming convention for Sofa.component.Utils

File: modules/Sofa/Component/Utils/config/Sofa.Component.Utils.h [hunk 4]
```diff
@@ -0,0 +1,48 @@
+/******************************************************************************
```
Comment (damienmarchal) [inline modules/Sofa/Component/Utils/config/Sofa.Component.Utils.h:1]: Please validate this new way to name config.h files

**Pull request merged**